### DISCUSSION TITLE
Dynamic class names

### DISCUSSION THREAD

**Discussion by borisghidaglia** [tailwindlabs/tailwindcss#9517] [Help]
Hello everyone,

I want to do a "cascading transition" (a transition where multiple elements play the same transition, but each one with an increasingly higher delay).

To implement this, I use the `delay` utility with arbitrary values.

The problem: in my code, I create classes dynamically. 
Thus, [as written in the doc](https://tailwindcss.com/docs/content-configuration#dynamic-class-names), Tailwind can't extract class names when doing this.

So, what is the idiomatic Tailwind way to deal with this, given the fact that I have too many elements to write each of them down by hand.

Thank you very much,

Boris

**Comment by wongjn** [comment]
You could:
- use the `style` attribute to apply the `transition-delay`. This is the simplest work-around if you don't need to rely on any variants. 
- use a CSS custom property as the value of the `transition-delay` class, such as `delay-[var(--delay)]` and then apply the concrete delay value as the value of the custom property, such as in a `style` attribute. This is useful if you need to apply the delay with any variants.

**Reply by borisghidaglia** [reply]
Thanks a lot for your answer!

Not sure I'm following you on the second point, could you expand a little bit?

**Reply by wongjn** [reply]
Best explained with an example:
```html
<div class="delay-[var(--delay)] md:delay-[calc(var(--delay)*2)]" style="--delay: 0s"></div>
<div class="delay-[var(--delay)] md:delay-[calc(var(--delay)*2)]" style="--delay: 1s"></div>
<div class="delay-[var(--delay)] md:delay-[calc(var(--delay)*2)]" style="--delay: 2s"></div>
<div class="delay-[var(--delay)] md:delay-[calc(var(--delay)*2)]" style="--delay: 3s"></div>
```
The `style` attribute is not subject to the Tailwind content scanning rules so you can place any dynamic values in there as needed.

**Reply by borisghidaglia** [reply]
Ah ok! Now I understand why you said "if you don't need to rely on any variants" in your first point (and also why and how the second point solves this pb).

Thank you for your time :)
